starknet/src/authenticators/eth_tx.cairo

```diff
@@ -1,35 +1,34 @@
 use starknet::ContractAddress;
 use sx::utils::types::{Strategy, IndexedStrategy, Choice};
 
-#[abi]
-trait IEthTxAuthenticator {
-    #[external]
+#[starknet::interface]
+trait IEthTxAuthenticator<TContractState> {
     fn authenticate_propose(
+        ref self: TContractState,
         target: ContractAddress,
         author: ContractAddress,
         execution_strategy: Strategy,
         user_proposal_validation_params: Array<felt252>
     );
-    #[external]
     fn authenticate_vote(
+        ref self: TContractState,
         target: ContractAddress,
         voter: ContractAddress,
         proposal_id: u256,
         choice: Choice,
         user_voting_strategies: Array<IndexedStrategy>
     );
-    #[external]
     fn authenticate_update_proposal(
+        ref self: TContractState,
         target: ContractAddress,
         author: ContractAddress,
         proposal_id: u256,
         execution_strategy: Strategy
     );
-    #[l1_handler]
-    fn commit(from_address: felt252, sender_address: felt252, hash: felt252);
+// TODO: Should L1 handlers be part of the interface?
```

Comment:
Hmmm I originally thought "yes" but now I think it shouldn't because users will not be able to call this function anyway so I don't see the point of having it in the interface

Reply:
But only callable by a single address, enforced by the protcol itself so idk ?